modules/gojira-core/src/main/java/com/flipkart/gojira/core/aspects/GlobalProfileOrTestAspect.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *
+ * Copyright 2020 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.orglicensesLICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.flipkart.gojira.core.aspects;
+
+import com.flipkart.gojira.core.ProfileOrTestMethodInterceptor;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aspectj.lang.annotation.Aspect;
+
+/**
+ * Use this class in aop.xml to extend to while defining concrete-aspects for defining custom pointcuts.
+ */
+@Aspect
+public abstract class GlobalProfileOrTestAspect extends AOPAllianceAdapter {
```

Comment:
This won't be required. ProfileOrTestAspect will do. We bind method interceptor only for methods annotated with ProfileOrTest as done here - https://github.com/flipkart-incubator/gojira/blob/master/modules/gojira-core/src/main/java/com/flipkart/gojira/core/BindingsModule.java

Reply:
We also wanted to support interception without touching the source code (or by having to add annotations). Especially for intercepting library methods. Hence we have exposed an abstract class which they can extend in aop.xml file and configure custom pointcuts (example in aop.xml file in test)